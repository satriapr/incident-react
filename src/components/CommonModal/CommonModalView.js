import React from 'react';
import {
  Button,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogContentText,
  DialogActions,
  Paper
} from '@mui/material';
import PropTypes from 'prop-types';

// Common Modal Component
const CommonModal = ({
  isOpen,
  handleClose,
  handleConfirm,
  title,
  description,
  isConfirmationButtonVisible
}) => {
  return (
    <Dialog
      open={isOpen}
      onClose={handleClose}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
      maxWidth="lg"
    >
      <Paper sx={{ p: 1 }}>
        <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-description">
            {description}
          </DialogContentText>
        </DialogContent>
        {isConfirmationButtonVisible && (
          <DialogActions>
            <Button onClick={handleClose}>Cancel</Button>
            <Button onClick={handleConfirm} variant="contained">
              Confirm
            </Button>
          </DialogActions>
        )}
      </Paper>
    </Dialog>
  );
};

export default CommonModal;

CommonModal.defaultProps = {
  isOpen: false,
  handleClose: () => {},
  title: '',
  description: '',
  isConfirmationButtonVisible: false,
  handleConfirm: () => {}
};

CommonModal.propTypes = {
  isOpen: PropTypes.bool,
  handleClose: PropTypes.func,
  title: PropTypes.string,
  description: PropTypes.string,
  isConfirmationButtonVisible: PropTypes.bool,
  handleConfirm: PropTypes.func
};
