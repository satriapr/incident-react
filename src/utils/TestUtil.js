import { shallow } from 'enzyme';
import * as React from 'react';

const singleSnapTest = (wrapper, description) => {
  test(description, () => {
    expect(wrapper).toMatchSnapshot();
  });
};

const testSnapshots = (Component, configs) => {
  describe('snapshots', () => {
    configs.forEach((config) => {
      const { props, description, state } = config;
      const wrapper = shallow(<Component {...props} />);
      if (state) wrapper.setState(state);
      singleSnapTest(wrapper, description);
    });
  });
};

export { testSnapshots };
