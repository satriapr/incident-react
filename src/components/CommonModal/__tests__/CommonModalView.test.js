import CommonModal from '../CommonModalView';
import { testSnapshots } from '../../../utils/TestUtil';

describe('CommonModal Component', () => {
  testSnapshots(CommonModal, [
    {
      description: 'CommonModal props',
      props: {
        isOpen: false,
        handleClose: () => {},
        title: '',
        description: '',
        isConfirmationButtonVisible: false,
        handleConfirm: () => {}
      }
    }
  ]);
});
