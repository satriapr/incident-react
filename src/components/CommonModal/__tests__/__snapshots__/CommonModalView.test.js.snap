// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonModal Component snapshots CommonModal props 1`] = `ShallowWrapper {}`;
